pkg/api/vm/handler.go

```diff
@@ -717,6 +719,48 @@ func (h *vmActionHandler) createVMImages(templateVersion *harvesterv1.VirtualMac
 	return nil
 }
 
+// Since the LH volume creation and replica scheduling are asynchronous
+// Even the CreateVolume csi call success, the replica scheduling may still fail
+// However for the sc with VolumeBindingMode is Immediate
+// we could check annotation for corresponding error if it exists
+func (h *vmActionHandler) checkAttachable(pvc *corev1.PersistentVolumeClaim) error {
+	// Even Harvester specify default StorageClass,
+	// we check the sc existence for robustness
+	if pvc.Spec.StorageClassName == nil {
+		return fmt.Errorf("volme %v with empty StorageClass is not permitted", pvc.Name)
+	}
```

Comment:
similar with L738, prefer skip the checking and log warning, return nil

Reply:
Since we could not have nil StorageClassName, how about keeping this checking for robustness as in the previous discussion https://github.com/harvester/harvester/pull/3869#discussion_r1194516279 ?